Support Wallet RPC protocol.

Chralu opened the pull request.

samuelmanzanera commented on README.md:
````diff
@@ -10,10 +10,46 @@ Official Archethic Javascript library for Node and Browser.
+```js
+import Archethic from "archethic";
+
+const archethic = new Archethic("ws://localhost:12345") // Endpoint is the Archethic Wallet RPC server
````
What happens for the other queries such as `getTransactionIndex`, `getOracleData` ?
I didn't see any forward or leverage of the `getEndpoint()` 
Does it mean it will require two archethic instances, one with the json rpc connection and one with the direct endpoint ?

Chralu replied:
`getEndpoint()` gets automatically resolved when calling `Archethic.connect()`.

That way, that `libjs` evolution is fully retrocompatible.

samuelmanzanera replied:
Ok

redDwarf03 commented:
Could you implement Failure management please like dart lib
```

  factory RPCFailure.unsupportedMethod() => const RPCFailure(
        code: -32601,
        message: 'Unsupported method.',
      );
  factory RPCFailure.timeout() => const RPCFailure(
        code: 5001,
        message: 'Operation timeout.',
      );
  factory RPCFailure.connectivity() => const RPCFailure(
        code: 4901,
        message: 'Connectivity issue.',
      );
  factory RPCFailure.consensusNotReached() => const RPCFailure(
        code: 5002,
        message: 'Validation consensus not reached',
      );
  factory RPCFailure.invalidParams() => const RPCFailure(
        code: -32602,
        message: 'Invalid parameters',
      );
  factory RPCFailure.invalidTransaction() => const RPCFailure(
        code: 5003,
        message: 'Invalid transaction',
      );
  factory RPCFailure.invalidConfirmation() => const RPCFailure(
        code: 5006,
        message: 'Invalid node confirmation.',
      );
  factory RPCFailure.insufficientFunds() => const RPCFailure(
        code: 5004,
        message: 'Insufficient funds.',
      );
  factory RPCFailure.serviceNotFound() => const RPCFailure(
        code: 5007,
        message: 'Service not found.',
      );
  factory RPCFailure.userRejected() => const RPCFailure(
        code: 4001,
        message: 'User rejected operation',
      );
  factory RPCFailure.unknownAccount() => const RPCFailure(
        code: 5005,
        message: 'Unknown account.',
      );
  factory RPCFailure.other() => const RPCFailure(
        code: 5000,
        message: 'Technical error',
      );
```

Chralu commented:
> Could you implement Failure management please like dart lib
> 
> ```
> 
>   factory RPCFailure.unsupportedMethod() => const RPCFailure(
>         code: -32601,
>         message: 'Unsupported method.',
>       );
>   factory RPCFailure.timeout() => const RPCFailure(
>         code: 5001,
>         message: 'Operation timeout.',
>       );
>   factory RPCFailure.connectivity() => const RPCFailure(
>         code: 4901,
>         message: 'Connectivity issue.',
>       );
>   factory RPCFailure.consensusNotReached() => const RPCFailure(
>         code: 5002,
>         message: 'Validation consensus not reached',
>       );
>   factory RPCFailure.invalidParams() => const RPCFailure(
>         code: -32602,
>         message: 'Invalid parameters',
>       );
>   factory RPCFailure.invalidTransaction() => const RPCFailure(
>         code: 5003,
>         message: 'Invalid transaction',
>       );
>   factory RPCFailure.invalidConfirmation() => const RPCFailure(
>         code: 5006,
>         message: 'Invalid node confirmation.',
>       );
>   factory RPCFailure.insufficientFunds() => const RPCFailure(
>         code: 5004,
>         message: 'Insufficient funds.',
>       );
>   factory RPCFailure.serviceNotFound() => const RPCFailure(
>         code: 5007,
>         message: 'Service not found.',
>       );
>   factory RPCFailure.userRejected() => const RPCFailure(
>         code: 4001,
>         message: 'User rejected operation',
>       );
>   factory RPCFailure.unknownAccount() => const RPCFailure(
>         code: 5005,
>         message: 'Unknown account.',
>       );
>   factory RPCFailure.other() => const RPCFailure(
>         code: 5000,
>         message: 'Technical error',
>       );
> ```

Done.

@samuelmanzanera  I removed errors obfuscation and added constants for known error codes.

Chralu commented:
@samuelmanzanera Rebase done.
